nengo/connection.py

```diff
@@ -119,14 +119,16 @@ class ConnectionFunctionParam(FunctionParam):
 
     def __set__(self, conn, function):
         if function is None:
-            function_info = self.Info(function=None, size=None)
+            function_info = FunctionInfo(function=None, size=None)
```

Comment:
What was the reason for separating out `FunctionInfo`? If I recall, it used to be separate but was put together in a somewhat recent PR... was this accidentally added in the rebase, or added back intentionally?